File: SoftUni Lessons/Python Development/Python Basics April 2019/Lessons and Problems/09 - While Loop Exercise/02. Exam Preparation .py
allowed_bad_scores = int(input())

tasks_counter = 0
bad_score_counter = 0
sum_score = 0
last_task = None

while bad_score_counter < allowed_bad_scores:
    task = input()
    if task == "Enough":
        sum_score = sum_score / tasks_counter
        print(f"Average score: {sum_score:.2f}")
        print(f"Number of problems: {tasks_counter}")
        print(f"Last problem: {last_task}")
        break
    score = int(input())
    sum_score += score
    tasks_counter += 1
    if score <= 4:
        bad_score_counter += 1
    last_task = task

if bad_score_counter == allowed_bad_scores:
    print(f"You need a break, {allowed_bad_scores} poor grades.")

File: SoftUni Lessons/JavaScript Development/JavaScript Basics March 2020/14. УПРАЖНЕНИЕ ЦИКЛИ - ЧАСТ 2/02. Exam Preparation.js
function examPreparation(data) {
    let limitBadScores = Number(data.shift());
    let badScores = 0;

    let totalScore = 0;
    let countSubjects = 0;
    let lastTask = undefined;

    while ("Lector says Enough or TOO many bad scores") {
        let subject = data.shift();
        if (subject === 'Enough') {
            console.log(`Average score: ${(totalScore / countSubjects).toFixed(2)}`);
            console.log(`Number of problems: ${countSubjects}`);
            console.log(`Last problem: ${lastTask}`);
            break;
        }
        let score = Number(data.shift());
        if (score <= 4) {
            badScores += 1;
            if (badScores === limitBadScores) {
                console.log(`You need a break, ${badScores} poor grades.`);
                break;
            }
        }
        totalScore += score;
        countSubjects += 1;
        lastTask = subject;
    }    
}

examPreparation([3, 'Money', 6, 'Story', 4, 'Spring Time', 5, 'Bus', 6, 'Enough']);
// Expected Output:
// Average score: 5.25
// Number of problems: 4
// Last problem: Bus

examPreparation([2, 'Income', 3, 'Game Info', 6, 'Best Player', 4]);
// Expected Output:
// You need a break, 2 poor grades.